src/bootstrap/src/lib.rs

```diff
@@ -1866,6 +1866,9 @@ pub fn generate_smart_stamp_hash(dir: &Path, additional_input: &str) -> String {
         .map(|o| String::from_utf8(o.stdout).unwrap_or_default())
         .unwrap_or_default();
 
+    eprintln!("Computing stamp for {dir:?}");
+    eprintln!("Diff output: {diff:?}");
```

Comment:
Good work on the caching speed-up. I expect the ad-hoc `eprintln` will need to be removed before we PR to upstream.

Reply:
I assume that once upstreamed we will generally change this CI setup significantly, but for now they were left extra s.t. we can try to find the reason if llvm again surprisingly is getting rebuild, so marking as solved for now.